packages/java-edition/src/mcfunction/mcdocAttributes.ts

```diff
@@ -0,0 +1,85 @@
+import * as core from '@spyglassmc/core'
+import { localize } from '@spyglassmc/locales'
+import * as mcdoc from '@spyglassmc/mcdoc'
+import * as mcf from '@spyglassmc/mcfunction'
+import { getItemSlotsArgumentValues } from './common/index.js'
+import {
+	EntitySelectorAtVariable,
+	EntitySelectorNode,
+	ScoreHolderNode,
+} from './node/argument.js'
+import * as parser from './parser/index.js'
+
+export function registerMcdocAttributes(
+	meta: core.MetaRegistry,
+	rootTreeNode: mcf.RootTreeNode,
+) {
+	mcdoc.runtime.registerAttribute(meta, 'command', () => undefined, {
+		// TODO: validate slash
+		// TODO: fix completer inside commands
+		stringParser: () => {
+			return mcf.command(rootTreeNode, parser.argument)
+		},
+	})
+	mcdoc.runtime.registerAttribute(meta, 'text_component', () => undefined, {
+		stringParser: () =>
+			makeInfallible(
+				parser.jsonParser('::java::server::util::text::Text'),
+				localize('text-component'),
+			),
+	})
+	mcdoc.runtime.registerAttribute(meta, 'objective', () => undefined, {
```

Comment:
I thought we implemented those using command arguments (objectuve, team, score_holder, selector and item_slot)? Was there an issue with this? Shouldn't it be possible to re-use the general mock function from mcfunction here as well?

Reply:
* Some command arguments require properties. Having them as separate attributes makes it easier to validate
* Shorter attributes makes the mcdoc source more readable
* I refactored the handler method to accept a parser, so they can be registered very compact here

Reply:
> * I refactored the handler method to accept a parser, so they can be registered very compact here

I would say, it is still some overhead (and more cases may occur in the future), and it would be really nice to have a command_argument attribute anyway for custom mcdoc, especially considering macros.